File: school/summer2013/example_openmp/c/41_offload.c
#include <stdio.h>

int main()
{
	long i, sum=0;
#pragma offload target(mic)
	#pragma omp parallel for reduction(+:sum)
	for(i=1; i<=N; i++)
		sum += i;

	printf("sum = %ld\n", sum);
}

File: summer2013/openmp/fortran/41_offload.f90
program sum_offload
    use omp_lib
    implicit none
    integer(8) :: i, total_sum = 0

!dir$ offload begin target(mic)
!$omp parallel do reduction (+:total_sum)
    do i=1,N
        total_sum = total_sum + i
    end do
!dir$ end offload

    print *, 'sum = ', total_sum
end
